File: Backend_Springboot/cab-booking/src/main/java/com/wheelshare/cabbooking/entity/Ride.java
package com.wheelshare.cabbooking.entity;

import jakarta.persistence.*;
import lombok.*;
import java.time.LocalDateTime;

@Entity
@Table(name = "rides")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Ride {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer rideId;

    // We can store just the ID or the relationship. Keeping relationship is better for JPA.
    // In .NET it had CustomerId and DriverId explicitly.
    
    @ManyToOne
    @JoinColumn(name = "customer_id", nullable = false)
    private User customer;

    @ManyToOne
    @JoinColumn(name = "driver_id")
    private Driver driver;

    private Double sourceLat;
    private Double sourceLng;
    private String sourceAddress;

    private Double destinationLat;
    private Double destinationLng;
    private String destinationAddress;

    private Double distanceKm;
    private Double fare;
    private Double finalFare;

    @Column(length = 20, nullable = false)
    private String rideStatus; // REQUESTED, ACCEPTED, STARTED, COMPLETED, CANCELLED

    @Builder.Default
    private LocalDateTime requestedAt = LocalDateTime.now();
    
    private LocalDateTime acceptedAt;
    private LocalDateTime completedAt;
}

File: Backend_DotNet/WheelShare_dotNet/WheelShare_dotNet/Models/Ride.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace WheelShare_dotNet.Models
{
    [Table("Rides")]
    public class Ride
    {

        [Key]
        public int RideId { get; set; }

        [Required]
        public int CustomerId { get; set; }

        public int? DriverId { get; set; }

        public double SourceLat { get; set; }
        public double SourceLng { get; set; }
        
        [MaxLength(500)]
        public string SourceAddress { get; set; }

        public double DestinationLat { get; set; }
        public double DestinationLng { get; set; }
        
        [MaxLength(500)]
        public string DestinationAddress { get; set; }

        public double DistanceKm { get; set; }
        public decimal Fare { get; set; }

        public decimal? FinalFare { get; set; }

        [Required, MaxLength(20)]
        public string RideStatus { get; set; }
        // Requested | Accepted | Started | Completed | Cancelled

        public bool IsCarpool { get; set; } = false;
        public int PassengerCount { get; set; } = 1;
        public int? ParentRideId { get; set; }

        public DateTime RequestedAt { get; set; } = DateTime.UtcNow;
        public DateTime? AcceptedAt { get; set; }
        public DateTime? StartedAt { get; set; }
        public DateTime? CompletedAt { get; set; }

        [ForeignKey("CustomerId")]
        public User Customer { get; set; }

        [ForeignKey("DriverId")]
        public Driver Driver { get; set; }
    }
}
